File: Shared/Blog.Contracts/V1/Responses/ErrorResponse.cs
namespace Blog.Contracts.V1.Responses
{
    using System.Collections.Generic;

    /// <summary>
    /// Error response.
    /// </summary>
    public class ErrorResponse
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="ErrorResponse"/> class.
        /// </summary>
        public ErrorResponse() { }

        /// <summary>
        /// Initializes a new instance of the <see cref="ErrorResponse"/> class.
        /// </summary>
        /// <param name="error">The error.</param>
        public ErrorResponse(ErrorModel error)
        {
            Errors.Add(error);
        }

        /// <summary>
        /// Gets or sets the errors.
        /// </summary>
        /// <value>
        /// The errors.
        /// </value>
        public List<ErrorModel> Errors { get; set; } = new List<ErrorModel>();
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Responses/ErrorResponse.fs
namespace Blog.FSharp.Contracts.V1.Responses

open System.Collections.Generic
open Blog.FSharp.Contracts

/// <summary>
/// Error response.
/// </summary>
type ErrorResponse() =
    /// <summary>
    /// Gets or sets the errors.
    /// </summary>
    member val Errors: List<ErrorModel> = List<ErrorModel>() with get, set

    /// <summary>
    /// Initializes a new instance of the <see cref="ErrorResponse"/> class with one error.
    /// </summary>
    /// <param name="error">The error.</param>
    new (error: ErrorModel) as this =
        ErrorResponse() then
        this.Errors.Add(error)